src/d3d11/d3d11_query.h

```diff
@@ -96,7 +96,7 @@ namespace dxvk {
     
   private:
     
-    D3D11Device* const m_device;
+    const Com<D3D11Device> m_device;
```

Comment:
Are we sure this should be ``Com<D3D11Device>`` and not ``Com<D3D11DeviceChild, false>`` for a private ref?

Reply:
> Are we sure this should be `Com<D3D11Device>` and not `Com<D3D11DeviceChild, false>` for a private ref?

I suggested it similar to how it looks like for, e. g., D3D11Buffer, any reason it should be different? I might have missed something here though, can't say I really went deep into dxvk ref counting while thought this is rather obvious case. If this is actually more involved maybe worth dropping this request in favour of a better solution.

Reply:
I'd imagine D3D11Buffer using public refs is probably wrong too. I will test this later today.

Reply:
D3D11Buffer inherits from ID3D11DeviceChild. Would be very weird if that didnt hold a public reference imo.

Reply:
I've just added the test to Wine tests for CreateBuffer specifically and tested on Windows, looks like the buffer holds a visible reference to device: https://gist.github.com/gofman/18b57f2e5907fc04cee60ff8e9462626

Reply:
Alright then, this lgtm!